sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -329,7 +329,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   /** Returns all the files (not directories) in a directory, recursively. */
   private def listFilesRecursively(path: File): Seq[File] = {
     val (dirs, files) = path.listFiles().partition(_.isDirectory)
-    files ++ dirs.flatMap(listFilesRecursively)
+    val filteredFiles = files.filter (_.getName matches validTestFileExtensionsRegEx)
```

Comment:
Let's avoid postfix syntax.
Not that it matters, but it is probably a little more efficient to make the regex a regex above with `.r` and then match with that pattern here.

Reply:
@srowen Thanks.. Sure... I will make the change. I wanted to confirm one thing.. so initially i had it as a regex as i thought, it may be easier for us to add extensions other than `.sql` in the future. But if we are going to stay with `.sql`, should we just do a `endsWith` check instead ? WDYT ?

Reply:
All sounds fine. The regex could match more patterns later, or could be many regexes. Or could indeed just check for suffixes for now for simplicity and make it more complicated later.